pubtools/_pulp/tasks/push/items/base.py

```diff
@@ -410,8 +410,21 @@ def ensure_uploaded(self, ctx, repo_f=None):
 
         upload_tasks = f_flat_map(repo_f, self.upload_to_repo)
 
-        # TODO: should we check here and raise if refresh didn't find that we're present?
-        return f_flat_map(upload_tasks, lambda _: self.with_pulp_refreshed(ctx.client))
+        # Helper to verify that we've really got into at least one repo as a result
+        # of the upload.
+        def asserting_uploaded_ok(item):
```

Comment:
minor: should these methods just be "assert_uploaded/updated", or are those name taken?

Reply:
It would be unusual for an 'assert' function or statement to return a value.  In this case we need a function which does a check on a value, but then also returns it.  The "ing" suffix is meant to indicate this, like: "given some value, give me the same value back, while asserting that some condition is satisfied".